basket/main.py

```diff
@@ -46,12 +71,26 @@ class Basket(object):
     # gets out of control.
     os = os
     tarfile = tarfile
-    urllib = urllib
+    urllib = urllib2
     zipfile = zipfile
     err = sys.stderr
     out = sys.stdout
     root = os.environ.get('BASKET_ROOT') or os.path.expanduser('~/.basket')
 
+    # Support http_proxy environment
+    # because of usage xmlrpclib and urllib we need to support both mechanisms
+    if os.environ.get('http_proxy'):
+        # Extract http_proxy from environment
+        proxy = ''.join(re.findall(r'[\d\.]+:?', os.environ.get('http_proxy')))
```

Comment:
The regexp looks odd to me but if it works for you, I guess it must be fine. :) Would you mind adding a comment as to why we just don't use the whole environment variable?